examples/scenarioSunlineFiltersCompare.py

```diff
@@ -0,0 +1,352 @@
+        scSim.AddModelToTask(simTaskName, css_residual_data_log)
+        css_measurement_noise_std = module.getCssMeasurementNoiseStd()
+
+    module.cssDataInMsg.subscribeTo(cssConstelation.constellationOutMsg)
+    module.cssConfigInMsg.subscribeTo(cssConstMsg)
+
+    scSim.AddModelToTask(simTaskName, navLog)
+    scSim.AddModelToTask(simTaskName, filtLog)
+
+    #
+    #   initialize Simulation
+    #
+    scSim.InitializeSimulation()
+
+    #
+    #   configure a simulation stop time and execute the simulation run
+    #
+    scSim.ConfigureStopTime(simulationTime)
+
+    # Time the runs for performance comparisons
+    start_time = time.time()
+    scSim.ExecuteSimulation()
+    end_time = time.time() - start_time
+
+    #
+    #   retrieve the logged data
+    #
+    def addTimeColumn(time, data):
+        return np.transpose(np.vstack([[time], np.transpose(data)]))
+
+    # Get messages that will make true data
+    timeAxis = dataLog.times()
+    OutSunPos = addTimeColumn(timeAxis, sunLog.PositionVector)
+    Outr_BN_N = addTimeColumn(timeAxis, dataLog.r_BN_N)
+    OutSigma_BN = addTimeColumn(timeAxis, dataLog.sigma_BN)
+    Outomega_BN = addTimeColumn(timeAxis, dataLog.omega_BN_B)
+
+    # Get the filter outputs through the messages
+    stateLog = addTimeColumn(timeAxis, filtLog.state[:, range(numStates)])
+    if FilterType == 'uKF':
+        postFitLog = addTimeColumn(timeAxis, filtLog.postFitRes[:, :8])
+    else:
+        postFitLog = addTimeColumn(css_residual_data_log.times(), css_residual_data_log.postFits[:, :8])
+    covarLog = addTimeColumn(timeAxis, filtLog.covar[:, range(numStates*numStates)])
+    # obsLog = addTimeColumn(timeAxis, filtLog.numObs)
+
+    sHat_B = np.zeros(np.shape(OutSunPos))
+    sHatDot_B = np.zeros(np.shape(OutSunPos))
+    for i in range(len(OutSunPos[:,0])):
+        sHat_N = (OutSunPos[i,1:] - Outr_BN_N[i,1:])/np.linalg.norm(OutSunPos[i,1:] - Outr_BN_N[i,1:])
+        dcm_BN = rbk.MRP2C(OutSigma_BN[i,1:])
+        sHat_B[i,0] = sHatDot_B[i,0]= OutSunPos[i,0]
+        sHat_B[i,1:] = np.dot(dcm_BN, sHat_N)
+        sHatDot_B[i,1:] = - np.cross(Outomega_BN[i,1:], sHat_B[i,1:] )
+
+    expected = np.zeros(np.shape(stateLog))
+    expected[:,0:4] = sHat_B
+
+    #   plot the results
+    #
+    errorVsTruth = np.copy(stateLog)
+    errorVsTruth[:,1:] -= expected[:,1:]
+
+    Fplot.StateErrorCovarPlot(errorVsTruth, covarLog, FilterType, show_plots, saveFigures)
+    Fplot.StatesVsExpected(stateLog, covarLog, expected, FilterType, show_plots, saveFigures)
+    Fplot.PostFitResiduals(postFitLog, css_measurement_noise_std, FilterType, show_plots, saveFigures)
+    # Fplot.numMeasurements(obsLog, FilterType, show_plots, saveFigures)
+
+    if show_plots:
+        plt.show()
+
+    # close the plots being saved off to avoid over-writing old and new figures
+    plt.close("all")
+
+
+    # each test method requires a single assert method to be called
```

Comment:
:small_red_triangle_down: Proof read all comments and ensure they make sense and add information that is not already obvious from the code.